src/utils/generateSummary.ts

```diff
@@ -0,0 +1,21 @@
+import { ElmJsDefinition } from "../contracts/ElmJsDefinition";
+import { Summary } from "../contracts/Summary";
+import { keyForSummaryType } from "./keyForSummaryType";
+import { rangeSize } from "./rangeSize";
+
+export function summaryGenerator(
```

Comment:
All output formats are based on this generator as the source of truth for what is to be shown.